### ISSUE TITLE
[Android] `MenuAnchor` items read with Talkback even when hidden

### ISSUE THREAD

**Issue by camsim99** [flutter/flutter#185784]
Not sure this is expected behavior, so filing just in case...

When running the [`a11y_assessments`](https://github.com/flutter/flutter/blob/master/dev/a11y_assessments) app, when I 

1. Open the `MenuAnchor` menu with Talkback
2. Close the menu with Talkback
3. Navigate to the right with Talkback 

I see that the menu items are still read aloud. I would expect those items not be read aloud if the menu is hidden, but please correct me if I'm wrong.

https://github.com/user-attachments/assets/fcf83768-88fc-4b7f-9102-fc1848f2ff32

**Comment by chunhtai** [comment]
This may be related to https://github.com/flutter/flutter/pull/185391.

Looking at the semantics tree, the items were removed from traversal tree when menu is closed, but for some reason the android embedding still has them. Can't tell if it is the framework side didn't to send the update or the android embedding retains the items incorrectly

cc @QuncCccccc

**Comment by jesswrd** [comment]
@QuncCccccc Do you think your PR from above will fix this issue?

**Comment by QuncCccccc** [comment]
I think this might be a different issue. The PR is to ensure missing nodes are visited so they aren't removed incorrectly, but this issue demo seems show that there are extra elements that should have been removed. ~Let me do a quick check.~ I did a quick check, the PR https://github.com/flutter/flutter/pull/185391 won't fix this. This should be a separate issue.

**Comment by QuncCccccc** [comment]
I think the issue should be on the framework side. When a traversal child is detached (menu closed), we should mark its traversal parent dirty so that the parent will be sent for update. Assigned to myself.
